strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/repositories/ArtifactRepository.java

```diff
@@ -0,0 +1,161 @@
+                                       Pageable pagination)
+    {
+        Page<Artifact> result = queries.findMatching(lastAccessedDate, minSizeInBytes, pagination);
+
+        return new PageImpl<>(EntityTraversalUtils.reduceHierarchy(result.toList()), pagination, result.getTotalElements());
+    }
+
+    public Boolean artifactEntityExists(String storageId,
+                                        String repositoryId,
+                                        String path)
+    {
+        return Optional.ofNullable(queries.artifactEntityExists(storageId, repositoryId, path)).orElse(Boolean.FALSE);
+    }
+
+    public Boolean artifactExists(String storageId,
+                                  String repositoryId,
+                                  String path)
+    {
+        EntityTraversal<Vertex, Vertex> t = g().V()
+                                               .hasLabel(Vertices.GENERIC_ARTIFACT_COORDINATES)
+                                               .has("uuid", path)
+                                               .inE(Edges.ARTIFACT_HAS_ARTIFACT_COORDINATES)
+                                               .otherV()
+                                               .hasLabel(Vertices.ARTIFACT)
+                                               .has("storageId", storageId)
+                                               .has("repositoryId", repositoryId)
+                                               .has("artifactFileExists", true);
+        return t.hasNext();
+    }
+
+    public Artifact findOneArtifact(String storageId,
+                                    String repositoryId,
+                                    String path)
+    {
+        org.carlspring.strongbox.storage.repository.Repository repository = configurationManager.getRepository(storageId, repositoryId);
+        
+        EntityTraversal<Vertex, Artifact> t = g().V()
+                                                 .hasLabel(Vertices.GENERIC_ARTIFACT_COORDINATES)
+                                                 .has("uuid", path)
+                                                 .inE(Edges.ARTIFACT_HAS_ARTIFACT_COORDINATES)
+                                                 .otherV()
+                                                 .hasLabel(Vertices.ARTIFACT)
+                                                 .has("storageId", storageId)
+                                                 .has("repositoryId", repositoryId)
+                                                 .map(artifactAdapter.fold(Optional.ofNullable(repository)
+                                                                           .map(org.carlspring.strongbox.storage.repository.Repository::getLayout)
+                                                                           .map(ArtifactLayoutLocator.getLayoutByNameEntityMap()::get)
+                                                                           .map(ArtifactLayoutDescription::getArtifactCoordinatesClass)));
```

Comment:
Cosmetic: Indentation.
```suggestion
                                                 .map(artifactAdapter.fold(Optional.ofNullable(repository)
                                                                                   .map(org.carlspring.strongbox.storage.repository.Repository::getLayout)
                                                                                   .map(ArtifactLayoutLocator.getLayoutByNameEntityMap()::get)
                                                                                   .map(ArtifactLayoutDescription::getArtifactCoordinatesClass)));
```